allAnagrams/allAnagramAlternatives.js

```diff
@@ -0,0 +1,79 @@
+// Recursive:
```

Comment:
Sarah, this is fantastic. allAnagrams is such a plaguing algorithm to complete. I hate it. haha. You've solved it recursively and with a depth first search solution. I'm very impressed!